WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift

```diff
@@ -553,6 +579,107 @@ import CoreData
         
         try coreDataStore.saveIfNeeded(moc: backgroundContext)
     }
+    
+    // TODO: Grey
+    private func populateViewCountSlide(report: CDYearInReviewReport, backgroundContext: NSManagedObjectContext, project: WMFProject?) throws {
+        guard let iosFeatureConfig = developerSettingsDataController.loadFeatureConfig()?.ios.first,
+              let _ = iosFeatureConfig.yir(yearID: targetConfigYearID) else {
+            throw WMFYearInReviewDataControllerError.missingRemoteConfig
+        }
+
+        guard let userID else { return }
+        
+        let encodedUserId = "#" + String(userID)
+        let languageCode = project?.languageCode
+        
+        fetchEditViews(project: project, userId: encodedUserId, language: languageCode ?? "en") { result in
+            switch result {
+            case .success(let totalViews):
+                guard let slides = report.slides as? Set<CDYearInReviewSlide> else {
+                    return
+                }
+                
+                for slide in slides {
+                    guard let slideID = slide.id else {
+                        continue
+                    }
+                    
+                    switch slideID {
+                    case WMFYearInReviewPersonalizedSlideID.viewCount.rawValue:
+                        do {
+                            let encoder = JSONEncoder()
+                            slide.data = try encoder.encode(totalViews)
+                            
+                            if totalViews > 0 {
+                                slide.display = true
```

Comment:
I think we'll run into threading issues because in this network call completion block, you are no longer on the Core Data thread, and updating Core Data models from that could cause crashes.

When we worked on your edit count slide, we determined it works better when we first make the network call separately to get the count, then ask the Core Data thread to update its report:

```
if result.needsEditingPopulation == true {
            if let username {
                let edits = try await fetchEditCount(username: username, project: primaryAppLanguageProject)
                try await backgroundContext.perform { [weak self] in
                    try self?.populateEditingSlide(edits: edits, report: report, backgroundContext: backgroundContext)
                }
            }
}
```

Can you follow that same pattern for this slide? You'll need to update `fetchEditViews` to be `async` instead of using completion blocks.

Also just a note, when I add `-com.apple.CoreData.ConcurrencyDebug 1` as a launch argument, I get complaints, but I get them even in `main` 😬. I can look into that tomorrow.

Reply:
Update: Luckily it was just a small fix for the `main` threading complaint, I pushed a commit fixing that. Now the launch argument complains in this area. Let's see if it reworking it to the edit count slide pattern fixes that, thanks!

![Screenshot 2024-12-05 at 7 51 47 PM](https://github.com/user-attachments/assets/160aae5f-96e0-4e90-84f2-63b346e40d37)